src/compiler/crystal/compiler.cr

```diff
@@ -344,17 +340,14 @@ module Crystal
           cc = CC
         end
 
-        if object_name
-          object_name = %('#{object_name}')
-        else
-          object_name = %("${@}")
```

Comment:
I suppose *here* I'm dropping the "advantage" of using `args`. That's the one case.

At the same time I'm removing an arbitrary code execution with a singular `object_name`